rasa/importers/autoconfig.py

```diff
@@ -0,0 +1,241 @@
+
+        missing_keys = _get_missing_config_keys(config)
+        keys_to_configure = _get_unspecified_autoconfigurable_keys(config)
+
+        if keys_to_configure:
+            config = _auto_configure(config, keys_to_configure)
+            _dump_config(config, config_file_path, missing_keys, keys_to_configure)
+
+    else:
+        logger.debug("No configuration file was provided to the TrainingDataImporter.")
+        config = {}
+
+    return config
+
+
+def _get_unspecified_autoconfigurable_keys(config: Dict[Text, Any]) -> List[Text]:
+    return [k for k in CONFIG_AUTOCONFIGURABLE_KEYS if not config.get(k)]
+
+
+def _get_missing_config_keys(config: Dict[Text, Any]) -> List[Text]:
+    return [k for k in CONFIG_KEYS if k not in config.keys()]
+
+
+def _auto_configure(
+    config: Dict[Text, Any], keys_to_configure: List[Text]
+) -> Dict[Text, Any]:
+    """Complete a config by adding automatic configuration for the specified keys.
+
+    Args:
+        config: The provided configuration.
+        keys_to_configure: Keys to be configured automatically (e.g. `policies`).
+
+    Returns:
+        The resulting configuration including both the provided and the automatically
+        configured keys.
+    """
+    import pkg_resources
+
+    if keys_to_configure:
+        logger.debug(
+            f"The provided configuration does not contain the key(s) "
+            f"{common_utils.transform_collection_to_sentence(keys_to_configure)}. "
+            f"Running automatic configuration for them now."
+        )
+
+    default_config_file = pkg_resources.resource_filename(
+        __name__, "default_config.yml"
+    )
+
+    default_config = io_utils.read_config_file(default_config_file)
+
+    config = copy.deepcopy(config)
+    for key in keys_to_configure:
+        config[key] = default_config[key]
+
+    return config
+
+
+def _dump_config(
+    config: Dict[Text, Any],
+    config_file_path: Text,
+    missing_keys: List[Text],
+    auto_configured_keys: List[Text],
+) -> None:
+    """Dump the automatically configured keys into the config file.
+
+    The configuration provided in the file is kept as it is (preserving the order of
+    keys and comments).
+    For keys that were automatically configured, an explanatory comment is added and the
+    automatically chosen configuration is added commented-out.
+    If there are already blocks with comments from a previous auto configuration run,
+    they are replaced with the new auto configuration.
+
+    Args:
+        config: The configuration including the automatically configured keys.
+        config_file_path: The file into which the configuration should be dumped.
+        missing_keys: Keys that need to be added to the config file.
+        auto_configured_keys: Keys for which a commented out auto configuration section
+                              needs to be added to the config file.
+    """
+    config_as_expected = _is_config_file_as_expected(
```

Comment:
very defensive 😄

Reply:
Ikr 😄 I was thinking about just letting it blow up cause it shouldn't ever happen, but there's no harm in keeping the check